source/common/filter/config_discovery_impl.h

```diff
@@ -102,12 +102,15 @@ class DynamicFilterConfigProviderImpl : public DynamicFilterConfigProviderImplBa
   }
 
   void onConfigRemoved(Config::ConfigAppliedCb applied_on_all_threads) override {
+    const absl::optional<Envoy::Http::FilterFactoryCb> default_config =
+        default_configuration_ ? absl::make_optional(factory_cb_(default_configuration_.value()))
+                               : absl::nullopt;
     tls_.runOnAllThreads(
-        [config = default_configuration_](OptRef<ThreadLocalConfig> tls) { tls->config_ = config; },
-        [this, applied_on_all_threads]() {
+        [config = default_config](OptRef<ThreadLocalConfig> tls) { tls->config_ = config; },
+        [this, default_config, applied_on_all_threads]() {
```

Comment:
I am skeptical if the captured `this` could survive when the main_callback is invoked.
Any theory to support `this` will outlive?

Reply:
I'm not sure, although is this out of the scope of this PR?

Reply:
yeah I'm skeptical too - the same thing might happen as grpc access loggers (currently being fixed).

Reply:
@mathetake can you share a reference here for my benefit? Thanks.

Reply:
https://github.com/envoyproxy/envoy/pull/18081 and https://github.com/envoyproxy/envoy/pull/18067 - basically I am worried about the case where DynamicFilterConfigProviderImpl has been deleted before this callback would be called on all threads. Then `current_config_` (which I imagine is used in `applied_on_all_threads` below) would be invalid address by the time when the callback is called.

Reply:
Yeah, this seems a plausible concern. I'm tempted to merge this though if @tbarrella wants to take this to a followup PR, since I don't see anything in this PR that makes the existing situation worse.